advntr/vntr_finder.py

```diff
@@ -366,23 +366,52 @@ def check_if_pacbio_read_spans_vntr(self, sema, read, length_distribution, spann
         sema.release()
 
     def check_if_pacbio_mapped_read_spans_vntr(self, sema, read, length_distribution, spanning_reads):
-        flanking_region_size = 100
-        region_start = self.reference_vntr.start_point - flanking_region_size
-        region_end = self.reference_vntr.start_point + self.reference_vntr.get_length()
-        if read.get_reference_positions()[0] < region_start and read.get_reference_positions()[-1] > region_end:
+        hmm_flanking_region_size = 100
+        min_flanking_bp = 10
+        vntr_start = self.reference_vntr.start_point
+        vntr_end = self.reference_vntr.start_point + self.reference_vntr.get_length()
+
+        region_start = vntr_start - hmm_flanking_region_size
+
+        first_aligned_position = read.get_reference_positions()[0]
+        last_aligned_position = read.get_reference_positions()[-1]
+        if first_aligned_position <= vntr_start - min_flanking_bp and vntr_end + min_flanking_bp < last_aligned_position:
             read_region_start = None
             read_region_end = None
-            for read_pos, ref_pos in enumerate(read.get_reference_positions()):
-                if ref_pos >= region_start and read_region_start is None:
-                    read_region_start = read_pos
-                if ref_pos >= region_end and read_region_end is None:
-                    read_region_end = read_pos
+
+            tr_spanning_bp = 0
+            left_flanking_bp = 0
+            right_flanking_bp = 0
+            for read_pos, ref_pos in enumerate(read.get_reference_positions(full_length=True)):
```

Comment:
Also do you think it would be helpful to store `get_reference_positions()` return value once and use it? Not sure what the complexity of computing this function is.

Reply:
I don't think it will be called over and over because it will be cashed.

Reply:
Great.